tasks/configure.yml

```diff
@@ -7,3 +7,27 @@
     mode: '0640'
   no_log: true
   with_items: '{{ restic_repos }}'
+
+- name: Check if the repo already exists
+  command: '{{ restic_install_path }}/restic snapshots'
+  environment:
+    RESTIC_REPOSITORY: '{{ item.url }}'
+    RESTIC_PASSWORD: '{{ item.password }}'
+  register: _repo_check
+  ignore_errors: True
+  changed_when: False
+  with_items: '{{ restic_repos }}'
+  when:
+    - item.init is defined
```

Comment:
Define it as `False` in `defaults/main.yml` and don't check if it is defined. That's what default values are for.